### PR TITLE
Update pipelineServer

### PR DESCRIPTION
**Pull request by pnaik1**
Closes: #1214  

## Description
Allow for a prefill from a Data Connection for reuse of data that have already setup
separating  the relationship between the Data Connection & the DSPA.

[Screencast from 2023-09-13 18-52-30.webm](https://github.com/opendatahub-io/odh-dashboard/assets/99238909/f77dff33-fa49-46b2-baa4-ca803ecb5284)

3) the view pipeline server
viewpipeline
 4) Modified the alert info in the configure pipeline modal
 
Screenshot from 2023-10-25 13-07-24
4) The dropdown with bucket whenever available

Screenshot from 2023-11-17 11-05-15

## How Has This Been Tested?

1. create a pipeline Server
2. check for the prefill dataconnection dropdown ( note the dropdown appears only when the data connection are present) 
3. check whether the pipeline server is configured correctly
4. check the view pipeline server whether the data are correctly displayed 

## Test Impact
no test coverage

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit tests & storybook for related changes)

If you have UI changes: 

- [x] Included any necessary screenshots or gifs if it was a UI change.
- [x] Included tags to the UX team if it was a UI/UX change.

After the PR is posted & before it merges:
- [x] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

**Comment by pnaik1** [comment]
cc @yannnz

**Comment by Gkrumbach07** [comment]
The dropdown is heavily squished and is missing padding visible in the mockups
image

**Comment by Gkrumbach07** [comment]
There are some UX inconsistencies in the color and weight of the non-hidden credentials
image

The keys should be bold and the text should not be grayed

File: frontend/src/concepts/pipelines/content/configurePipelinesServer/PipelineDropdown.tsx
Comment (Gkrumbach07) [inline frontend/src/concepts/pipelines/content/configurePipelinesServer/PipelineDropdown.tsx:101]: is there an empty state?

File: frontend/src/concepts/pipelines/content/configurePipelinesServer/PipelineDropdown.tsx
Comment (christianvogt) [inline frontend/src/concepts/pipelines/content/configurePipelinesServer/PipelineDropdown.tsx:98]: Seems there's some bad code in the `MenuGroup` when it comes to supplying custom titles. 

I think this would mimic the default styling:

```suggestion
import styles from '@patternfly/react-styles/css/components/Menu/menu';
import { css } from '@patternfly/react-styles';
...
            label={
              <h1 className={css(styles.menuGroupTitle)}>
                <KeyIcon /> Populate the form with credentials from your selected data connection
              </h1>
            }
```

File: frontend/src/pages/projects/dataConnections/const.ts
Comment (manaswinidas) [inline frontend/src/pages/projects/dataConnections/const.ts:48]: Can't we reuse `AWS_FIELDS`? This is a subset of `AWS_FIELDS` though.

File: frontend/src/pages/projects/dataConnections/const.ts
Comment (manaswinidas) [inline frontend/src/pages/projects/dataConnections/const.ts:98]: Can't we reuse `EMPTY_AWS_SECRET_DATA`? Again this is a subset of `EMPTY_AWS_SECRET_DATA`.

File: frontend/src/concepts/pipelines/content/configurePipelinesServer/ConfigurePipelinesServerModal.tsx
Comment (pnaik1) [reply frontend/src/concepts/pipelines/content/configurePipelinesServer/ConfigurePipelinesServerModal.tsx:113]: I discussed this with @yannnz (  the button is "Configure pipeline server", as as the model title.)

File: frontend/src/concepts/pipelines/context/PipelinesContext.tsx
Comment (pnaik1) [reply frontend/src/concepts/pipelines/context/PipelinesContext.tsx:169]: same here

File: frontend/src/concepts/pipelines/content/configurePipelinesServer/PipelineDropdown.tsx
Comment (pnaik1) [reply frontend/src/concepts/pipelines/content/configurePipelinesServer/PipelineDropdown.tsx:90]: no dont have a empty state here , the dropdown only gets rendered when there are data connections

File: frontend/src/pages/projects/dataConnections/const.ts
Comment (pnaik1) [reply frontend/src/pages/projects/dataConnections/const.ts:48]: AWS_FIELDS have the variables sightly change , here all the fields are required

File: frontend/src/pages/projects/dataConnections/const.ts
Comment (pnaik1) [reply frontend/src/pages/projects/dataConnections/const.ts:98]: To simplify the things are have separated them

File: frontend/src/concepts/pipelines/content/configurePipelinesServer/ObjectStorageSection.tsx
Comment (pnaik1) [reply frontend/src/concepts/pipelines/content/configurePipelinesServer/ObjectStorageSection.tsx:38]: Can we do it this way? taking single field in array form `<FieldList
            onUpdate={(update) => {
              setConfig({
                ...config,
                objectStorage: {
                  newValue: update,
                },
              });
            }}
            values={config.objectStorage.newValue}
            fields={[field]}
          />`

File: frontend/src/concepts/pipelines/content/configurePipelinesServer/PipelineDropdown.tsx
Comment (Gkrumbach07) [reply frontend/src/concepts/pipelines/content/configurePipelinesServer/PipelineDropdown.tsx:90]: I was able to get to the dropdown even when there was no data connections

**Review by Gkrumbach07 — COMMENTED** [review]
small nit

these are all greyed instead of a pure black. is this intended, because the mocks shwo them as black not grey

image

**Comment by Gkrumbach07** [comment]
> Just to confirm , so I can give the width properly
> I agree need some width for this , but i was wondering at the screenshot you shared is it looking too squished?
> My cluster is looking like this

Tested again and it looks like you fixed the padding issues and the width is correct now

**Comment by Gkrumbach07** [comment]
other than this comment [here](https://github.com/opendatahub-io/odh-dashboard/pull/1719#pullrequestreview-1631968200) it is good

/lgtm

**Comment by andrewballantyne** [comment]
As long as we use PF components and do not use custom styles... we are good to go. UX Mocks are NOT UI Mocks, they are not pixel perfect from a spacing, sizing, or colour perspective. Do not use those as a single source of truth for what the UI should look like -- follow PF component implementation at all points.

**Comment by andrewballantyne** [comment]
/hold
This is a feature request... feature freeze just occurred. 

@pnaik1 in the future consider posting your PR in a thread [like this](https://redhat-internal.slack.com/archives/C05SMJ09DD2/p1695843868937069). We'll work to improve our sprint tracking as time goes on.

**Comment by pnaik1** [comment]
> small nit
> 
> these are all greyed instead of a pure black. is this intended, because the mocks shwo them as black not grey
> 
> image

this is the  default css of menu description...

**Comment by xianli123** [comment]
@andrewballantyne I want to chime in because this issue has a relationship with the new request [#1711](https://github.com/opendatahub-io/odh-dashboard/issues/1711) and #1271. Should we also support the self-signed cert in the pipeline configuration part?  Thanks!

**Comment by andrewballantyne** [comment]
> @andrewballantyne I want to chime in because this issue has a relationship with the new request [#1711](https://github.com/opendatahub-io/odh-dashboard/issues/1711) and #1271. Should we also support the self-signed cert in the pipeline configuration part? Thanks!

@xianli123 probably not needed for the first pass. We need to support all kinds of Data Connection stuff first.

**Comment by andrewballantyne** [comment]
> [Screencast from 2023-09-13 18-52-30.webm](https://github.com/opendatahub-io/odh-dashboard/assets/99238909/f77dff33-fa49-46b2-baa4-ca803ecb5284)

@pnaik1 I think we want to try and put that info alert into the footer of the modal (you can scroll it out of view from this recording) and it popping in when the button enables is weird. The value of what you typed should not change the value of that message. We should definitely add it initially and permanently.

We don't need to put the info alert in the footer if we can have it at the top of the modal... @xianli123 thoughts?

**Comment by xianli123** [comment]
@andrewballantyne Yes, I agree. We should let users know it before they fill out the form. But I wonder if the alert is designed for the Database part or the whole config form.

**Comment by andrewballantyne** [comment]
> @andrewballantyne Yes, I agree. We should let users know it before they fill out the form. But I wonder if the alert is designed for the Database part or the whole config form.

Both sections cannot be edited after creation. It was intentionally immutable because there isn't solid support for reconciling the effort behind it. Changing these values requires a delete of the DSPA (server & all data) and recreate.

**Comment by xianli123** [comment]
Ok. It would be better to put the alert text on the top of the form. The alert will be more obvious than plain text. Does it make sense?

In addition, I found users can't view this page again after creation, so I think we could also tell users how to "edit" this server. It could be said that: `Changing values after creation requires a delete of the pipeline server and recreate.` The wording needs the help of the content designer to optimize. So @kaedward could you please take a look at the alert as shown below and optimize the wording of that? Thanks!

Screenshot 2023-10-19 at 11 29 16

**Comment by andrewballantyne** [comment]
> I found users can't view this page again after creation

@xianli123 they can from the Pipelines page and recently we [just updated it](https://github.com/opendatahub-io/odh-dashboard/issues/1542) so they can do it from the Projects page. 

There is a `view` option in both of these... and a delete option. But there is no real guidance; so I agree we should probably let them know upfront.

From the [mocks](https://www.sketch.com/s/6c72b6b2-2c32-45ff-b07c-381d9d8c8267/prototype/99D332C0-EA2D-4B94-8EFA-D3A46811BC6A/a/1F6F745B-3A7F-430B-B633-84B2D970BBB3).
image

**Comment by xianli123** [comment]
@andrewballantyne Ahh, I missed the Pipeline page. Thanks for sharing that. It looks good to have the same actions on the Project page as the [#1542](https://github.com/opendatahub-io/odh-dashboard/issues/1542) did. 

> I agree we should probably let them know upfront.

Let's go ahead with the alert in the creation form.

**Comment by kaedward** [comment]
@xianli123 For the alert text:

"Pipeline server configuration cannot be edited after creation. To use a different configuration after creation, delete the pipeline server and create a new one."

What do you think?

**Comment by xianli123** [comment]
Thanks @kaedward. It's more accurate. It's more accurate and more professional. I updated it. 
Screenshot 2023-10-25 at 11 23 33

**Comment by pnaik1** [comment]
@andrewballantyne does this pr still needs to be on hold?

**Comment by pnaik1** [comment]
@xianli123 made the changes , the screenshot is attached to the pr description

**Comment by andrewballantyne** [comment]
> @andrewballantyne does this pr still needs to be on hold?

@pnaik1 What is the reason for the hold? Is that reason no longer valid?

**Comment by pnaik1** [comment]
> > @andrewballantyne does this pr still needs to be on hold?
> 
> @pnaik1 What is the reason for the hold? Is that reason no longer valid?

The hold label  was not removed , so I thought to confirm before removing the label

**Review by xianli123 — APPROVED** [review]

**Comment by pnaik1** [comment]
@Gkrumbach07 @manaswinidas Can I get a final review for this?

**Comment by Gkrumbach07** [comment]
bucket is not pulled from the data connection or shown in the dropdown menu. is this intended?

**Comment by pnaik1** [comment]
> bucket is not pulled from the data connection or shown in the dropdown menu. is this intended?

@Gkrumbach07 referring to the Mock https://www.sketch.com/s/6c72b6b2-2c32-45ff-b07c-381d9d8c8267/a/JAamzrb , the dropdown doesnot have bucket in the list

**Comment by andrewballantyne** [comment]
If they have a bucket in their list, please use it. There is no reason to make them type that again.

File: frontend/src/concepts/pipelines/content/configurePipelinesServer/ObjectStorageSection.tsx
Comment (DaoDaoNoCode) [inline frontend/src/concepts/pipelines/content/configurePipelinesServer/ObjectStorageSection.tsx:74]: Got a typo at the end.
```suggestion
                    'Populate the form with credentials from your selected data connection.'
```

**Review by DaoDaoNoCode — CHANGES_REQUESTED** [review]
Just a small text issue, otherwise it looks good.

**Review by DaoDaoNoCode — APPROVED** [review]
/lgtm

**Comment by pnaik1** [comment]
@alexcreasy tested the image in the cluster and localy both are working fine

**Review by alexcreasy — APPROVED** [review]
Works well, looks good, thanks for rebasing and verifying this works deployed on a cluster.

/approve

**Pull request merged**
